src/lx/print/c.c

```diff
@@ -19,15 +19,16 @@
 #include <print/esc.h>
 
 #include "libfsm/internal.h" /* XXX */
+#include "libfsm/print/ir.h" /* XXX */
 
 #include "lx/ast.h"
 #include "lx/print.h"
 
 /* XXX: abstraction */
 int
-fsm_print_cfrag(FILE *f, const struct fsm *fsm,
+fsm_print_cfrag(FILE *f, const struct ir *ir, const struct fsm_options *opt,
```

Comment:
Once it's working with the IR, what is `struct fsm_options *opt` still needed for? Could that be stored within the IR instead?

Reply:
For various rendering options, like "always hex". I could duplicate those to an equivalent `struct ir_options` containing just the relevant subset perhaps - I actually tried splitting the .c files here such that they don't include any of the FSM structs at all. But I decided there wasn't any benefit, since this is all internal anyway.